docs/adr_1-implement_serverless_real_time.md

```diff
@@ -0,0 +1,139 @@
+&rarr; Hard to implement our authentication.
+* Too much refactor : using AppSync require to build an other backend from scratch. There are few pieces of code which could be kept.
+
+**Schema:**
+
+![AppSync schema](appsync_schema.png)
+
+### 3. Custom broker of Michalkvasnicak
+
+Use the package [aws-lambda-graphql](https://github.com/michalkvasnicak/aws-lambda-graphql)
+**Pro:** 
+
+* Solution designed to migrate our stack to AWS serverless :
+ The package is developed to setup a PubSub solution for subscriptions in GraphQL.
+* Seems to be the serverless way to implement real time, the solution is only made of serverless basic components : 
+Api GateWay, lambdas and dynamoDb
+
+**Cons:** 
+
+* Performances : latency between the mutation request and the reception of the subscribed message
+* Complex architecture &rarr; hard to understand, debug and maintain
+
+**Schemas:**
+
+![my schema of Michalkvasnicak solution](my_schema_of_Michalkvasnicak_solution.jpg)
+
+![Michalkvasnicak schema](michalkvasnicak_schema.svg)
+
+### 4. Develop our own custom broker from the solution 3
+
+Start from the [aws-lambda-graphql](https://github.com/michalkvasnicak/aws-lambda-graphql), remove unnecessary things and try to improve the performances. 
+
+**Pro:** 
+
+* It allows us to tailor-make a solution matching all the decision drivers.
+* Seems to be the serverless way to implement real time, the solution is only made of serverless basic components : 
+Api GateWay, lambdas and dynamoDb
+
+**Cons:** 
+
+* Long to develop
+* No certainty about the results
+* Complex architecture &rarr; hard to understand, debug and maintain
+ 
+**Schema:** 
+ 
+ current schema
+ ![custom broker schema](DynamoDb_pubsub_serverless.png)
+ 
+ ### 5. Serverless components : realtime-app
+ 
+ Use the package [serverless-components/realtime-app/](https://github.com/serverless-components/realtime-app/).
+ It's a Component of [Serverless Components](https://github.com/serverless/components).
+ 
+ The pitch of Serverless Components :
+ "It's a framework for provisioning and composing cloud services into higher-level abstractions, like features and applications. It depends on external Components, which are javascript libraries that can provision something, like infrastructure or a whole software feature built on multiple pieces of infrastructure." 
+ 
+ It's an overlay of the framework [Serverless](https://serverless.com/) 
+ 
+ **Pro:** 
+ 
+ * Solution designed to implement real time in serverless
+ * It implements [Serverless components](https://github.com/serverless/components) which make the solution easy to deploy :
+ a une command to create the AWS infrastructure (if it does not exist) and deploy new code on it.
+ 
+ **Cons:** 
+ * No mention of offline development
+ * All the code which deal with realtime data must be on an unique lambda &rarr; for us that means to deploy all our backend in one lambda
+ * New package without proper documentations
```

Comment:
What is a problem of New package ?

Reply:
There is no docs and probably some bugs : when I tested the package, there is a functionality I didn't manage to use, I don't know if I badly used the package our if it's a bug.